### PR TITLE
Fix instantiation of composite property ref

### PR DESCRIPTION
**Pull request by dreab8**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/EmbeddedAttributeMapping.java [hunk 1]
```diff
@@ -406,6 +406,7 @@ private static PropertyAccess getPropertyAccess(
 					true );
 		}
 		else {
+//			throw new NotYetImplementedFor6Exception();
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/EmbeddedAttributeMapping.java:409]: Unnecessary

**Comment by beikov** [comment]
Looks good overall, but AFAIU these changes are necessary for the `EmbeddableInitializer` to receive the fetch parent access, so that it can inject the state to it's "virtual instance" which is the entity itself, right? I guess we will need access to the fetch parent access anyway even if we model virtual embeddables differently, so I'm fine with the change.

**Comment by dreab8** [comment]
@beikov yeap we need access to the parent  to inject the state to it's "virtual instance" which is the entity itself

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/EmbeddedAttributeMapping.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/EmbeddedAttributeMapping.java:409]: yeap an oversight :) I'm going to delete it thanks

**Pull request merged**